sdmetrics/reports/single_table/diagnostic_report.py

```diff
@@ -0,0 +1,264 @@
+
+        out.write('DiagnosticResults:\n')
+        if len(self._results['SUCCESS']) > 0:
+            out.write('\nSUCCESS:\n')
+            for result in self._results['SUCCESS']:
+                out.write(f'{result}\n')
+
+        if len(self._results['WARNING']) > 0:
+            out.write('\nWARNING:\n')
+            for result in self._results['WARNING']:
+                out.write(f'{result}\n')
+
+        if len(self._results['DANGER']) > 0:
+            out.write('\nDANGER:\n')
+            for result in self._results['DANGER']:
+                out.write(f'{result}\n')
+
+    def generate(self, real_data, synthetic_data, metadata):
+        """Generate report.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+        """
+        metrics = list(itertools.chain.from_iterable(self.METRICS.values()))
+
+        for metric in tqdm.tqdm(metrics, desc='Creating report'):
+            metric_name = metric.__name__
+            try:
+                self._metric_results[metric_name] = metric.compute_breakdown(
+                    real_data, synthetic_data, metadata)
+
+                if 'score' in self._metric_results[metric_name]:
+                    self._metric_averages[metric_name] = self._metric_results[metric_name]['score']
+                else:
+                    avg_score, _ = aggregate_metric_results(self._metric_results[metric_name])
+                    self._metric_averages[metric_name] = avg_score
+
+            except IncomputableMetricError:
+                # Metric is not compatible with this dataset.
+                self._metric_results[metric_name] = {}
+
+        self._print_results()
+
+    def get_results(self):
+        """Return the diagnostic results.
+
+        Returns:
+            dict
+                The diagnostic results.
+        """
+        return copy.deepcopy(self._results)
+
+    def get_properties(self):
+        """Return the property score breakdown.
+
+        Returns:
+            pandas.DataFrame
+                The property score breakdown.
+        """
+        properties = {}
+        for prop, metrics in self.METRICS.items():
+            prop_scores = [self._metric_averages[metric.__name__] for metric in metrics]
+            properties[prop] = np.nanmean(prop_scores)
+
+        return properties
+
+    def get_visualization(self, property_name):
+        """Return a visualization for each score for the given property name.
+
+        Args:
+            property_name (str):
+                The name of the property to return score details for.
+
+        Returns:
+            plotly.graph_objects._figure.Figure
+                The visualization for the requested property.
+        """
```

Comment:
what's supposed to happen here?

Reply:
This method is implemented in a separate PR ([here](https://github.com/sdv-dev/SDMetrics/pull/238)).

I put in `pass` but I think `make lint` removed in.